Zend/zend_object_handlers.c

```diff
@@ -833,6 +828,16 @@ ZEND_API zval *zend_std_write_property(zend_object *zobj, zend_string *name, zva
 					variable_ptr = &EG(error_zval);
 					goto exit;
 				}
+				if (UNEXPECTED(prop_info->flags & ZEND_ACC_READONLY)) {
```

Comment:
I just realized that this is actually wrong. `if (UNEXPECTED(prop_info->flags & ZEND_ACC_READONLY)) {` should be done before `zend_verify_property_type`, because otherwise side-effects like `__toString()` will run even if the property cannot be assigned. `Z_PROP_FLAG_P(variable_ptr) &= ~IS_PROP_REINITABLE;` however should be run after. @kocsismate Can you adjust that?

Reply:
@iluuu1994 Yes, sure. I actually had the very same implementation initially for a few minutes, just to switch it to the current implementation. :)

Reply:
Adding a test should be simple enough. Create a class with `__toString` and assign to a readonly property that's already initialized.

Reply:
Yep, I have already created a similar test case locally :)